broker/tasks/letsencrypt.py

```diff
@@ -329,11 +336,18 @@ def cert_from_fullchain(fullchain_pem: str) -> str:
             )
             raise e
     except errors.ValidationError as e:
-            logger.error(
-                f"failed to retrieve certificate for {service_instance.domain_names} with errors {e.failed_authzrs}"
-            )
-            raise e
-
+        logger.error(
+            f"failed to retrieve certificate for {service_instance.domain_names} with errors {e.failed_authzrs}"
+        )
+        # if we fail validation, nuke the cert record and its challenges.
+        # this way, when we retry from the beginning, we won't try to reuse them
+        # the bad new is that we'll still retry this task a bunch of times before the pipeline fails
```

Comment:
The cert will still fail in future retries after eliminating the remnants from the failed attempt at renewing?  Do we just have to let it run its course, or is there something else that can be done?

Reply:
right - when it fails once, we nuke all this stuff, then on all the other retries it will continue failing (but differently - probably running into `Nonetype doesn't have attribute ...` ). We have a fixed limit on retries, so eventually we'll fail the whole pipeline. On a provision or update operation, this will cause the broker & CAPI to see this as a failure (which would've happened anyway).
Renewals are where this really matters. By cleaning these resources up, the next time cron triggers a new renewal pipeline for this instance, it'll get a fresh start. This means (unlike the existing case) if the error is intermittent (e.g. dns error, race condition, etc), we have hope of renewing the next time around.

Reply:
Thank you for explaining!